daal4py/sklearn/_config.py

```diff
@@ -0,0 +1,72 @@
+# ==============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Tools to expose sklearnex's config settings to daal4py level."""
+
+import threading
+
+_default_global_config = {
+    "target_offload": "auto",
+    "allow_fallback_to_host": False,
+}
+
+_threadlocal = threading.local()
+
+
+def _get_daal4py_threadlocal_config():
+    if not hasattr(_threadlocal, "d4p_global_config"):
+        _threadlocal.d4p_global_config = _default_global_config.copy()
+    return _threadlocal.d4p_global_config
+
+
+def _get_config():
```

Comment:
As I have comment it above, I will remove it from __init__. This was not intended to be of public iface.
This will be only internally used. 
From my opinion naming is fine in case of consistency for sklearnex config. I probably will reduce some part of this file, but the rest is fine

Reply:
outdated